src/librustc_typeck/diagnostics.rs

```diff
@@ -4092,6 +4092,110 @@ impl SpaceLlama for i32 {
 ```
 "##,
 
+E0527: r##"
+The number of elements in an array or slice pattern differed from the number of
+elements in the array being matched.
+
+Example of erroneous code:
+
+```compile_fail,E0527
+#![feature(slice_patterns)]
+
+fn main() {
+    let r = &[1, 2, 3, 4];
+    match r {
+        &[a, b] => { // error: pattern requires 2 elements but array
+                     //        has 4
+            println!("a={}, b={}", a, b);
+        }
+    }
+}
+```
+
+Ensure that the pattern is consistent with the size of the matched
+array. Additional elements can be matched with `..`:
+
+```ignore
```

Comment:
See my [earlier comment](https://github.com/rust-lang/rust/pull/34319#issuecomment-227732647):

> [...] the corrected code samples were failing tests due to the lack of the `#![feature(slice_patterns)]` attribute. I've marked them as `ignore` because I had deliberately left the attribute out because it's already in the erroneous code samples just above and I want to conserve vertical terminal space for people running `--explain`.